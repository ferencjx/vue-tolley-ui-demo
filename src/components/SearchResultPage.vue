<template>
<div>
    <h1>{{ msg }}</h1>
    <SearchResultHeader :resultsCount=resultsCount />
    <SearchResultList :results=results />
</div>
</template>

<script>
import Vue from 'vue'

import SearchResultList from './SearchResultList'
import SearchResultHeader from './SearchResultHeader'

import resultjson from '../mockdata/data.json'

export default {
  name: 'SearchResultPage',
  components: {
    SearchResultHeader,
    SearchResultList
  },
  data () {
    return searchResultList
  }
}

var searchResultList = new Vue({
  data: {
    msg: 'Search Result List',
    results: resultjson.results,
    resultsCount: resultjson.resultsCount
  }
})

</script>

<style scoped>

</style>
