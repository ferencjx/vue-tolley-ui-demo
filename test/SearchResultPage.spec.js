import { shallowMount } from '@vue/test-utils'

import SearchResultPage from '../src/components/SearchResultPage'

describe('SearchResultPage', () => {
  it('renders the page', () => {
    const wrapper = shallowMount(SearchResultPage)
    expect(wrapper).toMatchSnapshot()
  })
})
