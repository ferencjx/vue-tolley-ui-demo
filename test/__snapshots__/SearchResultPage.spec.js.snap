// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultPage renders the page 1`] = `
<div>
  <h1>Search Result List</h1>
  <searchresultheader-stub resultscount="250"></searchresultheader-stub>
  <searchresultlist-stub results="[object Object],[object Object],[object Object]"></searchresultlist-stub>
</div>
`;
